### PR TITLE
Replay mode

### PR DESCRIPTION
**Pull request by ivnminin**

### CODE REVIEW

**Review by ongrid — CHANGES_REQUESTED** [review]
Need detailed logging about what is happening
It should be startable as aiohttp app, not standalone

File: app.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+UPSTREAM = os.environ.get("UPSTREAM")
+
+@routes.view(r"/{URN:.*}")
+class MyView(web.View):
+
+    async def get(self):
+
+        if UPSTREAM:
+            print("UPSTREAM!")
```
Comment (ongrid) [inline app.py]: not informative. Add dateiled logging.

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+aiohttp-jinja2==1.4.2
+async-timeout==3.0.1
+attrs==20.3.0
+certifi==2020.11.8
+chardet==3.0.4
+idna==2.10
+Jinja2==2.11.2
+MarkupSafe==1.1.1
+multidict==5.0.2
+pkg-resources==0.0.0
```
Comment (ongrid) [inline requirements.txt]:
```
(venv) MacBook-Pro-kirill:mocksha kirill$ python -m aiohttp.web -H localhost -P 8080 mocksha
/Users/kirill/Prog/mocksha/venv/bin/python: Error while finding module specification for 'aiohttp.web' (ModuleNotFoundError: No module named 'aiohttp')
(venv) MacBook-Pro-kirill:mocksha kirill$ python -m pip install -r requirements.txt 
Collecting aiohttp==3.7.3
  Using cached aiohttp-3.7.3-cp39-cp39-macosx_10_14_x86_64.whl (648 kB)
Collecting aiohttp-debugtoolbar==0.6.0
  Using cached aiohttp_debugtoolbar-0.6.0-py3-none-any.whl (465 kB)
Collecting aiohttp-jinja2==1.4.2
  Using cached aiohttp_jinja2-1.4.2-py3-none-any.whl (11 kB)
Collecting async-timeout==3.0.1
  Using cached async_timeout-3.0.1-py3-none-any.whl (8.2 kB)
Collecting attrs==20.3.0
  Using cached attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Collecting certifi==2020.11.8
  Using cached certifi-2020.11.8-py2.py3-none-any.whl (155 kB)
Collecting chardet==3.0.4
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting idna==2.10
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting Jinja2==2.11.2
  Using cached Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
Collecting MarkupSafe==1.1.1
  Using cached MarkupSafe-1.1.1.tar.gz (19 kB)
Collecting multidict==5.0.2
  Using cached multidict-5.0.2-cp39-cp39-macosx_10_14_x86_64.whl (48 kB)
ERROR: Could not find a version that satisfies the requirement pkg-resources==0.0.0 (from -r requirements.txt (line 12)) (from versions: none)
ERROR: No matching distribution found for pkg-resources==0.0.0 (from -r requirements.txt (line 12))
WARNING: You are using pip version 20.2.3; however, version 20.3 is available.
You should consider upgrading via the '/Users/kirill/Prog/mocksha/venv/bin/python -m pip install --upgrade pip' command.
```

File: utils.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import os, re, yaml
+
+
+mocksha_app_dir = os.path.abspath(os.path.dirname(__file__))
+YAML_files_store = os.path.join(mocksha_app_dir, "YAML_files_store")
```
Comment (ongrid) [inline utils.py]: use `config_dir`
and the directory should be created if it doesn't exist.

File: app.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                del headers["Transfer-Encoding"]
+                headers["Content-Length"] = str(len(raw))
+
+            data.update({
+                "response": {
+                    "status": res.status,
+                    #TODO
+                    "headers": multidict_to_dict(response.headers),
+                    "content": {
+                        "body": raw,
```
Comment (ongrid) [inline app.py]: Why it's always binary?

```
response:
  content:
    body: !!binary |
      PCFET0NUWVBFIGh0bWw+PGh0bWwgbGFuZz0iZW4iPjxoZWFkPjxtZXRhIGNoYXJzZXQ9InV0Zi04
      Ij48bWV0YSBuYW1lPSJ2aWV3cG9ydCIgY29udGVudD0id2lkdGg9ZGV2aWNlLXdpZHRoLCBpbml0
      aWFsLXNjYWxlPTEsIHNocmluay10by1maXQ9bm8sIG1heGltdW0tc2NhbGU9MSwgdXNlci1zY2Fs
      YWJsZT1ubyI+PGxpbmsgcmVsPSJzaG9ydGN1dCBpY29uIiB0eXBlPSJpbWFnZS9zdmcreG1sIiBo
      cmVmPSJmYXZpY29uLjQ3YTcwZjNiLnN2ZyI+PGxpbmsgcmVsPSJzaG9ydGN1dCBpY29uIiB0eXBl
      PSJpbWFnZS9wbmciIGhyZWY9ImZhdmljb24uZjQ1MTM2OTIucG5nIj48dGl0bGU+QXJhZ29uPC90
```

**Comment by ongrid** [comment]
with `export UPSTREAM=http://206.81.31.11:5052` running the script

`requests.get('http://127.0.0.1:8080/eth/v1/node/version', headers = {'accept': 'application/json'})`

should produce plaintext JSON in the file

**Comment by ongrid** [comment]
the app shouldn't start if config_dir is not empty (with human readable log)

File: requirements.txt (hunk 2 above)
Comment (ivnminin) [reply requirements.txt]: According to https://github.com/pypa/pip/issues/4022, this is a bug resulting from Ubuntu providing incorrect metadata to pip.
`pip uninstall pkg-resources==0.0.0`
updated requirements.txt

**Pull request closed without merging**